### PR TITLE
Removing db dependency for MainSearcher

### PR DESCRIPTION
**Pull request by andrewconner**
**Please do not merge yet. Passes tests, but I would like review first.**

This was started by Yingjie. It ended up affecting a massive number of other classes. I added a good amount of additional testing and reorganized modules. I have tested in development mode with split services and it seems to find the bindings it needs.

### CODE REVIEW

**Comment by eishay** [comment]
wow! biggie

File: shoebox/app/com/keepit/common/analytics/EventListener.scala [hunk 1]
```diff
@@ -224,9 +227,21 @@ class SearchUnloadListenerImpl @Inject() (
...
+          }
+          searchClient.getSearchStatistics(uuid, queryString, userId, labeledUris).map { r =>
+            r.value.map { json =>
+              val event = Events.serverEvent(EventFamilies.SERVER_SEARCH, "search_statistics", json.as[JsObject])
+              persistEventProvider.get.persist(event)
+            }
+          }
+          log.info("search statistics persisted")
+        }
+ //       searchClient.persistSearchStatistics(uuid, queryString, userId, kifiClickedIds, googleClickedIds, kifiShownIds)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala]: why is it commented out?

Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala]: are we using search statistics? maybe we don't need it

Comment (andrewconner) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala]: We are using it.

File: shoebox/app/com/keepit/search/SearchServiceClient.scala [hunk 2]
```diff
@@ -55,15 +54,31 @@ class SearchServiceClientImpl(override val host: String, override val port: Int,
...
-        "kifiShown" -> kifiShown.map{_.id})
-    call((routes.SearchStatisticsController.persistSearchStatistics), json).map{r => ()}
+        "uriIds" -> uriIds,
+        "uriLabels" -> uriLabels
+        )
+    call(routes.SearchStatisticsController.getSearchStatistics, json).map{
+        r => r.json.as[JsArray]
+    }
   }
+//
```
Comment (eishay) [inline shoebox/app/com/keepit/search/SearchServiceClient.scala]: why the commenting out?

**Pull request merged**